### PR TITLE
[DOCS] Update Model Serving docs

### PR DESCRIPTION
**Pull request by davesh0812**
https://iguazio.atlassian.net/browse/ML-5770

This pr also fix the keras option

### CODE REVIEW

File: docs/serving/model-api.md [hunk 1]
```diff
@@ -76,7 +76,53 @@ Request body:
...
+     - Allows a list of lists for processing multiple data points simultaneously, 
+with each data point containing features of different types.
+     - Example: `[[feature1a, feature2a, feature3a, ...], [feature1b, feature2b, feature3b, ...], ..]`
+
+  - Note: If the user wants to send an image as an input, it should be sent as a list of RGB lists. 
+  This format is only accepted in the **batch** case. For example: `[[[[R1, G1, B1], [R2, G2, B2], ...]],...]`
+
+- **outputs:** Optional, requested output values.
+
+Let me know if there's anything else you'd like to add!
```
Comment (assaf758) [inline docs/serving/model-api.md]: Remove this line :-)

File: docs/serving/model-api.md [hunk 2]
```diff
@@ -76,7 +76,53 @@ Request body:
 - **model**: Model to select (for streaming protocols without URLs).
 - **data_url**: Option to load the `inputs` from an external file/s3/v3io/.. object.
 - **parameters**: Optional request parameters.
-- **inputs**: List of input elements (numeric values, arrays, or dictionaries).
+- **inputs**: Inputs for a model, where each data point should be provided as a list. 
+Each data point can be extracted from different features with varying types. 
+For convenience, users can also pass a list of lists to enable batch processing.
```
Comment (assaf758) [inline docs/serving/model-api.md]: Remove this line (redundant) as mentioned in "Batch input"

File: docs/serving/model-api.md [hunk 3]
```diff
@@ -76,7 +76,53 @@ Request body:
...
+  1. **Single Data Point Input:** 
+     - Accepts a list representing a single data point, which can include features of different types.
+     - Example: `[feature1, feature2, feature3, ...]`
```
Comment (assaf758) [inline docs/serving/model-api.md:83]: Any restrictions on the type of the individual feature?

File: docs/serving/model-api.md [hunk 4]
```diff
@@ -76,7 +76,53 @@ Request body:
...
+     - Example: `[[feature1a, feature2a, feature3a, ...], [feature1b, feature2b, feature3b, ...], ..]`
+
+  - Note: If the user wants to send an image as an input, it should be sent as a list of RGB lists. 
+  This format is only accepted in the **batch** case. For example: `[[[[R1, G1, B1], [R2, G2, B2], ...]],...]`
+
+- **outputs:** Optional, requested output values.
+
+Let me know if there's anything else you'd like to add!
+## infer_dict / predict_dict
+- This API is particularly helpful when the user does not remember the order of the features.
```
Comment (assaf758) [inline docs/serving/model-api.md]: Better start with description of what it is before following with the benefits

**Review by assaf758 — CHANGES_REQUESTED** [review]
Good stuff.
Some comments on the md

File: docs/serving/model-api.md (hunk 3 above)
Comment (davesh0812) [reply docs/serving/model-api.md:83]: the feature have to be serializable - add this sentence in 2 places.

File: docs/serving/model-api.md [hunk 5]
```diff
@@ -76,9 +76,54 @@ Request body:
...
+      "outputs" : [ $request_output, ... ] #optional
+    }
+
+- **id**: Unique Id of the request, if not provided a random value is provided.
+- **model**: Model to select (for streaming protocols without URLs).
+- **data_url**: Option to load the `inputs` from an external file/s3/v3io/.. object.
+- **parameters**: Optional request parameters.
+- **inputs**: Inputs for a model, where each data point should be provided as a dictionary. 
+Each data point can be extracted from different features with varying types, the feature have to be serializable. 
+This API support only batch mode.
```
Comment (jillnogold) [inline docs/serving/model-api.md]: This API only supports batch mode.

File: docs/tutorials/03-model-serving.ipynb [hunk 6]
```diff
@@ -352,7 +376,7 @@
     "<a id=\"deploy-serving\"></a>\n",
     "## Deploy the serving function\n",
     "\n",
-    "Deploy the serving function and use `invoke` to test it with the provided `sample`."
+    "Deploy the serving function and use `invoke` to test it with the provided `sample` by 2 API's `infer` & `infer_dict`."
```
Comment (jillnogold) [inline docs/tutorials/03-model-serving.ipynb]: "Deploy the serving function and use `invoke` to test it with the provided `sample` by using 2 API's `infer` & `infer_dict`."

File: docs/serving/model-api.md [hunk 7]
```diff
@@ -76,9 +76,54 @@ Request body:
 - **model**: Model to select (for streaming protocols without URLs).
 - **data_url**: Option to load the `inputs` from an external file/s3/v3io/.. object.
 - **parameters**: Optional request parameters.
-- **inputs**: List of input elements (numeric values, arrays, or dictionaries).
+- **inputs**: Inputs for a model, where each data point should be provided as a list. 
+Each data point can be extracted from different features with varying types, the feature have to be serializable.
+  1. **Single Data Point Input:**
```
Comment (jillnogold) [inline docs/serving/model-api.md]: - **Single Data Point Input:**

File: docs/serving/model-api.md [hunk 8]
```diff
@@ -76,9 +76,54 @@ Request body:
 - **model**: Model to select (for streaming protocols without URLs).
 - **data_url**: Option to load the `inputs` from an external file/s3/v3io/.. object.
 - **parameters**: Optional request parameters.
-- **inputs**: List of input elements (numeric values, arrays, or dictionaries).
+- **inputs**: Inputs for a model, where each data point should be provided as a list. 
+Each data point can be extracted from different features with varying types, the feature have to be serializable.
+  1. **Single Data Point Input:** 
+     - Accepts a list representing a single data point, which can include features of different types.
+     - Example: `[feature1, feature2, feature3, ...]`
+  2. **Batch Input:**
```
Comment (jillnogold) [inline docs/serving/model-api.md]: - **Batch Input:**

File: docs/serving/model-api.md [hunk 9]
```diff
@@ -76,9 +76,54 @@ Request body:
...
+    {
+      "id" : $string #optional,
+      "model" : $string #optional
+      "data_url" : $string #optional
+      "parameters" : $parameters #optional,
+      "inputs" : [ $request_input, ... ],
+      "outputs" : [ $request_output, ... ] #optional
+    }
+
+- **id**: Unique Id of the request, if not provided a random value is provided.
```
Comment (jillnogold) [inline docs/serving/model-api.md]: - **id**: Unique ID of the request. If not provided a random value is used.

File: docs/serving/model-api.md [hunk 10]
```diff
@@ -76,9 +76,54 @@ Request body:
...
+- **model**: Model to select (for streaming protocols without URLs).
+- **data_url**: Option to load the `inputs` from an external file/s3/v3io/.. object.
+- **parameters**: Optional request parameters.
+- **inputs**: Inputs for a model, where each data point should be provided as a dictionary. 
+Each data point can be extracted from different features with varying types, the feature have to be serializable.
```
Comment (jillnogold) [inline docs/serving/model-api.md]: Each data point can be extracted from different features with varying types, the features have to be serializable.

File: docs/serving/model-api.md [hunk 11]
```diff
@@ -76,9 +76,54 @@ Request body:
...
+    }
+
+- **id**: Unique Id of the request, if not provided a random value is provided.
+- **model**: Model to select (for streaming protocols without URLs).
+- **data_url**: Option to load the `inputs` from an external file/s3/v3io/.. object.
+- **parameters**: Optional request parameters.
+- **inputs**: Inputs for a model, where each data point should be provided as a dictionary. 
+Each data point can be extracted from different features with varying types, the feature have to be serializable. 
+This API supports only batch mode.
+1**Batch Input:**
```
Comment (jillnogold) [inline docs/serving/model-api.md]: - **Batch Input:**

**Review by jillnogold — APPROVED** [review]

**Review by assaf758 — APPROVED** [review]

**Pull request merged**